src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryProcessorChainBuilder.cs

```diff
@@ -104,8 +94,15 @@ public void Build()
                 }
 
                 telemetryProcessorsList.Add(linkedTelemetryProcessor);
-            }
 
+                // If a Processor also implements ITelemtryModule, We should Initialize that Module
+                if (linkedTelemetryProcessor is ITelemetryModule telemetryModule)
+                {
+                    telemetryModule.Initialize(this.configuration);
```

Comment:
We protect against re-initialization across thread boundaries in some of the `ITelemetryModule`s I've seen and worked on.  However, it is entirely possible that a 3rd party may not have done so. Would it be prudent to put this inside a try...catch block? (Please safely ignore this comment if there is an outer try catch I cannot see).

Reply:
That's a good point. We have no control over other implementations, and our own [TelemetryConfigurationFactory calls Initialize in a try/catch](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryConfigurationFactory.cs#L372-L379). I'll do the same here.